src/main/java/org/ibp/api/rest/inventory/manager/LotResource.java

```diff
@@ -198,12 +202,32 @@ public long getCount() {
 		@ApiParam("Request with fields to update and criteria to update") @RequestBody final LotUpdateRequestDto lotRequest) {
 
 		final BindingResult errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
-		this.inventoryCommonValidator.validateSearchCompositeDto(lotRequest.getSearchComposite(), errors);
-		final LotsSearchDto searchDTO = this.searchRequestDtoResolver.getLotsSearchDto(lotRequest.getSearchComposite());
 
-		final List<ExtendedLotDto> extendedLotDtos = this.lotService.searchLots(searchDTO, null);
-		if (lotRequest.getSearchComposite().getSearchRequest() == null) {
-			this.extendedLotListValidator.validateAllProvidedLotUUIDsExist(extendedLotDtos, lotRequest.getSearchComposite().getItemIds());
+		List<ExtendedLotDto> extendedLotDtos = null;
+
+		if ((lotRequest.getSingleInput() == null && lotRequest.getMultiInput() == null) || //
+		 	(lotRequest.getSingleInput() != null && lotRequest.getMultiInput() != null)) {
+			errors.reject("lot.update.invalid.input", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+		}
+
+		if (lotRequest.getSingleInput() != null) {
+			this.inventoryCommonValidator.validateSearchCompositeDto(lotRequest.getSingleInput().getSearchComposite(), errors);
+			final LotsSearchDto searchDTO =
+				this.searchRequestDtoResolver.getLotsSearchDto(lotRequest.getSingleInput().getSearchComposite());
+
+			extendedLotDtos = this.lotService.searchLots(searchDTO, null);
+			if (lotRequest.getSingleInput().getSearchComposite().getSearchRequest() == null) {
+				this.extendedLotListValidator
+					.validateAllProvidedLotUUIDsExist(extendedLotDtos, lotRequest.getSingleInput().getSearchComposite().getItemIds());
+			}
+		} else {
+			final List<String> lotUIDs =
```

Comment:
@cuenyad are we validating that the UIDs can't be repeated? How is the bahaviour in that case?

Reply:
@clarysabel I added a validateLotUUIDsDuplicated for this case.